friendships/tests.py

```diff
@@ -0,0 +1,84 @@
+# =================================================================================================
+#                                  All Rights Reserved.
+# =================================================================================================
+# File description:
+#           Friendship model unit tests
+#
+# =================================================================================================
+#    Date      Name                    Description of Change
+# 07-Sep-2021  Wayne Shih              Initial create
+# $HISTORY$
+# =================================================================================================
+
+
+from testing.testcases import TestCase
+from friendships.models import Friendship
+
+
+class FriendshipTest(TestCase):
+
+    def setUp(self):
+        self.user_1 = self.create_user(username='cavs_lbj23')
+        self.user_2 = self.create_user(username='lakers_lbj23')
+        self.user_3 = self.create_user(username='bulls_mj23')
+
+        self.friendship_1_3 = Friendship.objects.create(
+            from_user=self.user_1,
+            to_user=self.user_3
+        )
+        self.friendship_2_3 = Friendship.objects.create(
+            from_user=self.user_2,
+            to_user=self.user_3
+        )
+
+    def test_friendship_model_attributes(self):
+        self.assertEqual(hasattr(Friendship, 'id'), True)
+        self.assertEqual(hasattr(Friendship, 'from_user'), True)
+        self.assertEqual(hasattr(Friendship, 'from_user_id'), True)
+        self.assertEqual(hasattr(Friendship, 'to_user'), True)
+        self.assertEqual(hasattr(Friendship, 'to_user_id'), True)
+        self.assertEqual(hasattr(Friendship, 'created_at'), True)
+
+    def test_friendship_model(self):
+        friendships = Friendship.objects.all()
+        self.assertEqual(friendships.count(), 2)
+        # <Wayne Shih> 07-Sep-2021
+        # test order by '-created_at'
+        self.assertEqual(friendships.first().from_user, self.user_2)
+        self.assertEqual(friendships.first().to_user, self.user_3)
+
+        self.user_2.delete()
+        self.assertEqual(friendships.first().from_user, None)
+
+        self.assertEqual(hasattr(self.user_1, 'following_friendship_set'), True)
+        self.assertEqual(hasattr(self.user_1, 'follower_friendship_set'), True)
+
+    def test_friendship_meta(self):
+        self.assertEqual(len(Friendship._meta.index_together), 2)
+        self.assertEqual(len(Friendship._meta.unique_together), 1)
```

Comment:
It might be better to check if index_together and unique_together are the one we are expected also.

Reply:
Resolved as per suggestion